src/main/java/uk/gov/pay/products/resources/ProductResource.java

```diff
@@ -156,8 +159,26 @@ public Response updateProduct(@PathParam("gatewayAccountId") Integer gatewayAcco
                             .orElseGet(() -> Response.status(NOT_FOUND).build()));
     }
 
+    /**
+     * Update product using JSON patch. 
+     * 
+     * TODO: Updates from /v1/api/gateway-account/{gatewayAccountId}/products/{productExternalId}
+     * should be moved over to this endpoint and that API deprecated. This was not done at the time due to time constraints.
+     */
+    @PATCH
+    @Path("/v2/api/products/{productExternalId}")
```

Comment:
Any reason we can't use `/v1/api/products/{productExternalId}` since this is unique ?